File: module-5/thi/thi-be/src/main/java/com/example/thibe/service/BookService.java
package com.example.thibe.service;


import com.example.thibe.model.Book;
import com.example.thibe.repository.IBookRepository;
import org.springframework.beans.factory.annotation.Autowired;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
public class BookService implements IBookService {
    @Autowired
    IBookRepository iBookRepository;

    @Override
    public Page<Book> findAll(String name, Pageable pageable) {
        return iBookRepository.findBookByNameContaining(name,pageable);
    }

    @Override
    public Book findById(Integer id) {
        return iBookRepository.findById(id).orElse(null);
    }

    @Override
    public void update(Book book) {
        iBookRepository.save(book);
    }

    @Override
    public void delete(Integer id) {
        iBookRepository.deleteById(id);
    }
}

File: module-5/thi/thi-fe/src/service/bookService.js
import axios from "axios";

export const findAll = async ({ page, name}) => {
    return await axios.get(`http://localhost:8080/api/books?name=${name}&page=${page ? page : 0}`);
}
export const findById = async (id) => {
    return await axios.get(`http://localhost:8080/api/books/${id}`);
}

export const findAllType = async () => {
    return await axios.get(`http://localhost:8080/api/bookTypes`);
}

export const save = async (book) => {
    return await axios.post(`http://localhost:8080/api/books`, {...book});
}
export const update = async (book) => {
    return await axios.put(`http://localhost:8080/api/books/${book.id}`, {...book});
}

export const remove = async (id) => {
    return await axios.delete(`http://localhost:8080/api/books/${id}`);
}

